File: Packages/RideSharingFeature/Sources/RideSharingFeature/DesignSystem/LiiveIconCircle.swift
//  LiiveIconCircle.swift  ·  Liive Ride DS (SwiftUI)
//  Tinted circular icon badge. Mirrors components/core/IconCircle.

import SwiftUI

public struct LiiveIconCircle: View {
    public enum Color { case accent, success, warning, danger, info, neutral }
    let systemName: String
    var color: Color = .accent
    var size: CGFloat = LiiveControl.md
    var filled: Bool = false

    public init(
        systemName: String,
        color: Color = .accent,
        size: CGFloat = LiiveControl.md,
        filled: Bool = false
    ) {
        self.systemName = systemName; self.color = color; self.size = size; self.filled = filled
    }

    private var fg: SwiftUI.Color {
        switch color {
        case .accent: return LiiveColor.accent
        case .success: return LiiveColor.success
        case .warning: return LiiveColor.warning
        case .danger: return LiiveColor.danger
        case .info: return LiiveColor.info
        case .neutral: return LiiveColor.textSecondary
        }
    }
    private var solid: SwiftUI.Color {
        switch color {
        case .accent: return LiiveColor.accent
        case .success: return LiiveColor.success
        case .warning: return LiiveColor.warning
        case .danger: return LiiveColor.danger
        case .info: return LiiveColor.info
        case .neutral: return LiiveColor.fill
        }
    }
    private var tint: SwiftUI.Color {
        switch color {
        case .accent: return LiiveColor.accentTint
        case .success: return LiiveColor.successTint
        case .warning: return LiiveColor.warningTint
        case .danger: return LiiveColor.dangerTint
        case .info: return LiiveColor.infoTint
        case .neutral: return LiiveColor.fillTertiary
        }
    }

    public var body: some View {
        Image(systemName: systemName)
            .font(.system(size: size * LiiveIconCircleLayout.iconScale, weight: .semibold))
            .foregroundColor(filled ? LiiveIconCircleLayout.filledForegroundColor : fg)
            .frame(width: size, height: size)
            .background(filled ? solid : tint)
            .clipShape(Circle())
    }
}

private enum LiiveIconCircleLayout {
    static let iconScale: CGFloat = 0.45
    static let filledForegroundColor = SwiftUI.Color.white
}

File: android/app/src/main/java/com/liive/ride/designsystem/LiiveIconCircle.kt
//  LiiveIconCircle.kt  ·  Liive Ride DS (Compose)  ·  mirrors components/core/IconCircle
package com.liive.ride.designsystem

import androidx.annotation.DrawableRes
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material3.Icon
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.unit.Dp

enum class IconCircleColor { Accent, Success, Warning, Danger, Info, Neutral }

@Composable
fun LiiveIconCircle(
    @DrawableRes icon: Int,
    color: IconCircleColor = IconCircleColor.Accent,
    size: Dp = LiiveControl.md,
    filled: Boolean = false,
) {
    val c = LiiveTheme.colors
    val fg = when (color) {
        IconCircleColor.Accent -> c.accent; IconCircleColor.Success -> c.success
        IconCircleColor.Warning -> c.warning; IconCircleColor.Danger -> c.danger
        IconCircleColor.Info -> c.info; IconCircleColor.Neutral -> c.textSecondary
    }
    val solid = when (color) {
        IconCircleColor.Accent -> c.accent; IconCircleColor.Success -> c.success
        IconCircleColor.Warning -> c.warning; IconCircleColor.Danger -> c.danger
        IconCircleColor.Info -> c.info; IconCircleColor.Neutral -> c.fill
    }
    val tint = when (color) {
        IconCircleColor.Accent -> c.accentTint; IconCircleColor.Success -> c.successTint
        IconCircleColor.Warning -> c.warningTint; IconCircleColor.Danger -> c.dangerTint
        IconCircleColor.Info -> c.infoTint; IconCircleColor.Neutral -> c.fillTertiary
    }
    Box(
        Modifier.size(size).clip(CircleShape).background(if (filled) solid else tint),
        contentAlignment = Alignment.Center
    ) {
        Icon(
            painterResource(icon),
            contentDescription = null,
            tint = if (filled) LiiveIconCircleLayout.FilledForegroundColor else fg,
            modifier = Modifier.size(size * LiiveIconCircleLayout.IconScale)
        )
    }
}

private object LiiveIconCircleLayout {
    const val IconScale = 0.45f
    val FilledForegroundColor = Color.White
}
